cmd/kube-controller-manager/app/options/options_test.go

```diff
@@ -459,8 +461,18 @@ func TestAddFlags(t *testing.T) {
 	// insertion order is random.
 	sort.Sort(sortedGCIgnoredResources(expected.GarbageCollectorController.GCIgnoredResources))
 
-	if !reflect.DeepEqual(expected, s) {
-		t.Errorf("Got different run options than expected.\nDifference detected on:\n%s", cmp.Diff(expected, s))
+	// Ignore unexported fields when comparing.
```

Comment:
Which field?  Why can't we set it to match?  We can make custom equalities if necessary (for time for instance)

Reply:
I guess https://pkg.go.dev/k8s.io/apimachinery/pkg/util/diff#ObjectGoPrintSideBySide is the way these days.